lib/foreman_maintain/concerns/os_facts.rb

```diff
@@ -0,0 +1,58 @@
+module ForemanMaintain
+  module Concerns
+    module OsFacts
+      OS_RELEASE_FILE = '/etc/os-release'.freeze
+
+      def facts
+        facts = {}
+        tr_map = { '"' => '', "\n" => '' }
+        rexp = Regexp.union(tr_map.keys)
+        file = File.open(OS_RELEASE_FILE)
```

Comment:
I'd always prefer a block with open so you're sure it's closed. While you're unlikely to leak file descriptors in maintain, for daemons it's a real risk.

Reply:
Added block